src/enigma2/Epg.cpp

```diff
@@ -0,0 +1,269 @@
+#include "Epg.h"
+
+#include <regex>
+
+#include "../client.h"
+#include "../Enigma2.h"
+#include "utilities/Logger.h"
+#include "utilities/WebUtils.h"
+
+#include "util/XMLUtils.h"
+#include "p8-platform/util/StringUtils.h"
+
+using namespace enigma2;
+using namespace enigma2::data;
+using namespace enigma2::extract;
+using namespace enigma2::utilities;
+
+Epg::Epg (enigma2::Channels &channels, enigma2::ChannelGroups &channelGroups, enigma2::extract::EpgEntryExtractor &entryExtractor)
+      : m_channels(channels), m_channelGroups(channelGroups), m_entryExtractor(entryExtractor)
+{
+  InitialiseEpgReadyFile();
+}
+
+void Epg::InitialiseEpgReadyFile()
+{
+  m_writeHandle = XBMC->OpenFileForWrite(INITIAL_EPG_READY_FILE.c_str(), true);
+  XBMC->WriteFile(m_writeHandle, "Y", 1);
+  XBMC->CloseFile(m_writeHandle);  
+}
+
+bool Epg::IsInitialEpgCompleted()
+{
+  m_readHandle = XBMC->OpenFile(INITIAL_EPG_READY_FILE.c_str(), 0);
+  byte buf[1];
```

Comment:
@phunkyfish this break windows, see https://jenkins.kodi.tv/view/Addons/job/kodi-pvr/job/pvr.vuplus/job/master/2/execution/node/191/log/